rfd/0174-kube-rbac-bootstrap.md

```diff
@@ -29,26 +29,79 @@ and we will not allow them to be provisioned. Although other resources might be
+Teleport will translate permissions defined in that field into Kubernetes RBAC definitions and will automatically create roles and bindings
+on the target Kubernetes clusters (as specified by the labels). If `namespaces` field includes '*' Teleport will create ClusterRole/ClusterRoleBinding
+pair on the Kubernetes cluster, otherwise Role/Binding pair will be created in the each listed namespace. 
+
+Subject for the bindings will be a group named `teleport:*RoleName*`, e.g. for a Teleport role `kube-role` the resulting 
+group name will be `teleport:kube-role`. This will allow Kube service to setup correct impersonation group headers based on the roles available to the user.
+
+If `kubernetes_permissions` field is set in the `allow` part of the Teleport role, fields `kubernetes_resources`, `kubernetes_users` and `kubernetes_groups` 
+should not be set, we will return an error if user will try to create a role mixing those fields. In the future we might relax this rule.
+
+The field `kubernetes_permissions` will be forbidden from appearing in the `deny` part of the role definition since we are trying to keep its meaning close to
+native Kubernetes RBAC, which can only allow access. Although the field `kubernetes_resource` could appear in the `deny` part of the role, 
+even if `kubernetes_permissions` is set in the `allow` part, giving user more flexibility using Teleport-level permissions check.
+
+Pattern matching for resource names that is supported by the `kubernetes_resources` field will not be supported (same as in native Kubernetes RBAC)
+by the `kubernetes_permissions` in the
+first implementation, but we will be able to add it later. Not allowing pattern matching, though limiting functionality compared to the `kubernetes_resources` field,
+will allow us to translate permissions defined in Teleport 1-to-1 into Kubernetes RBAC definitions. Since this feature is targeted primarily for the new users of 
+Teleport and/or users with less complicated RBAC setups, this should not be a critical issue. If we add pattern matching later, it will need to rely on 
+additional checks at the Teleport level - the way it currently works for the `kubernetes_resources`.
+
+Kubernetes RBAC will be auto-provisioned using the KubeProvisions functionality described in the next section of this RFD. For the best UX any change to
+a role that contains `kubernetes_permissions` will trigger auto-provisioning immediately to reduce latency between the moments when role was created 
+or changed and when clients can use Kubernetes cluster with the updated RBAC.
+
+Introducing `kubernetes_permissions` field will make enrolling into Teleport and starting using Kubernetes clusters
+a much simpler process, since user will be able to setup RBAC completely in Teleport if they want. We will also be able
 to amend our documentation to make our guides simpler, which will also help users to start with Teleport more easily. 
 
 #### UX
```

Comment:
I think the biggest UX challenge is that this adds yet another way of configuring Kuberntes RBAC. Most users also don't understand how Teleport and kube RBAC interact.

IIUC, a user wanting to control kube RBAC would have 3 approaches:
- create users/groups with restricted rights and use `kubernetes_labels` + `kubernetes_users/groups`
- create admin RBAC and use `kubernetes_labels` + `kubernetes_resources` to restrict what users can do
- use the new shiny `kubernetes_permissions` and let the Teleport agent create the RBAC you need

User were already confused when we had 2 ways of configuring RBAC, but now they will have 3 😅 
To reduce the confusion, I think we should have a dedicated documentation page comparing the 3 approaches and advising users to use `kubernetes_permissions` if they don't have an existing setup.

Reply:
>use the new shiny kubernetes_permissions and let the Teleport agent create the RBAC you need

This won't cover all cases that we currently support like with deny clauses, resource filtering... 
I wonder if it will be simple to explain how everything works and when you should use one or two together.